OpenRA.Mods.Common/Lint/CheckTargetHealthRadius.cs

```diff
@@ -59,32 +69,77 @@ public void Run(Action<string> emitError, Action<string> emitWarning, Ruleset ru
 						if (!warhead.ValidTargets.Overlaps(targetable))
 							continue;
 
-						if (healthTraits.Where(x => x.Shape.OuterRadius.Length > warhead.TargetSearchRadius.Length).Any())
-							emitError("Actor type `{0}` has a health radius exceeding the victim scan radius of a CreateEffectWarhead on `{1}`!"
-								.F(actorInfo.Key, weaponInfo.Key));
+						var warheadScanRadius = warhead.TargetSearchRadius != WDist.Zero ? warhead.TargetSearchRadius.Length : largestHealthRadius;
+						if (healthTraits.Where(x => x.Shape.OuterRadius.Length > warheadScanRadius).Any())
+							emitError("Actor type `{0}` has a health radius exceeding the victim scan radius of `{1}` a CreateEffectWarhead on `{2}`!"
+								.F(actorInfo.Key, warheadScanRadius.ToString(), weaponInfo.Key));
 					}
 
 					var bullet = weaponInfo.Value.Projectile as BulletInfo;
+					var instant = weaponInfo.Value.Projectile as InstantHitInfo;
 					var missile = weaponInfo.Value.Projectile as MissileInfo;
 					var areabeam = weaponInfo.Value.Projectile as AreaBeamInfo;
+					var laserzap = weaponInfo.Value.Projectile as LaserZapInfo;
```

Comment:
This is getting horrible.  Can we please implement an interface that exposes `TargetExtraSearchRadius` and `Blockable` without having to hardcode types here?